### DISCUSSION TITLE
extendTheme not working for Alert component

### DISCUSSION THREAD

**Discussion by georgeweiler** [chakra-ui/chakra-ui#4739] [Q&A] [answered]
Can someone help me understand why my theme extension is not applying to the Alert component? I tested this in a brand new CRA repo using `yarn create react-app my-app --template @chakra-ui/typescript`. Here's my `App.tsx`:

```

const customTheme = extendTheme(
    withDefaultColorScheme({
      colorScheme: "orange",
      components: ["Badge"],
    }),
    withDefaultColorScheme({
      colorScheme: "purple",
      components: ["Alert", "Button"],
    }),
)

export const App = () => (
  <ChakraProvider theme={customTheme}>
        <VStack spacing={8}>
            <Badge>I should be orange</Badge>

            <Button>I should be purple</Button>

            <Alert>
                <AlertIcon />
                <AlertDescription>
                    I Should be purple too!
                </AlertDescription>
            </Alert>

        </VStack>

  </ChakraProvider>
)
```

and the rendered output:

![Screen Shot 2021-09-20 at 10 21 46 AM](https://user-images.githubusercontent.com/15269424/134046110-0786e0ae-68a3-4f55-8a14-c02a057cea2a.png)

you can see that the Alert component does not pick up the default color scheme from the custom theme. It also ignores the other theme extensions such as `baseStyle`, `defaultProps`, etc.

What am I missing?

**Comment by primos63** [comment, marked as answer]
Why are you applying the color scheme to an alert? Is your intention to have it that color no matter the status? If so, using `withDefaultColorScheme` is not going to work. If, however, your intention is only to configure the color for the default status, `info`, then I'd say this is a bug.

`colorScheme` is either passed in as a prop or it is determined from the status, which here defaults to `info` which makes `colorScheme='blue'`.
The theme for the `Alert` component contains `defaultProps` that after applying `withDefaultColorScheme` is
```javascript
const defaultProps = {
  variant: "subtle",
  colorScheme: "purple",
}
```
`useMultiStyleConfig`  is called using the `colorScheme='blue'`. Within `useMultiStyleConfig` all of the styles and props are merged. Since `colorScheme` is passed to the function, it is the last property merged and overwrites the `colorScheme` from the theme. The application of the `colorScheme` in `defaultProps` only applies to the default status `info`. If the status is passed as a prop and it is not `info` then the color should be according to the `STATUSES` color array. For example, `<Alert status='warning'>` would result in a `colorScheme` of orange being applied and not the value from `defaultProps`.

To fix this case, the `colorScheme` should be taken from the theme when the status is `info` and the `colorScheme` prop is not specified.

- `<Alert>` uses default status of `info` and `colorScheme` from the default props section of Alert's theme.
- `<Alert colorScheme='pink'>` the color scheme will be pink no matter the status.
- `<Alert status='warning'>` the color scheme will be the default for status warning, orange.

Alert.tsx:
```jsx
export const Alert = forwardRef<AlertProps, "div">((props, ref) => {
  const { status = "info", ...rest } = omitThemingProps(props)
  const colorScheme = props.colorScheme ?? STATUSES[status].colorScheme

  const styles = useMultiStyleConfig("Alert", { ...props, colorScheme })
  ...
}
```

**Reply by georgeweiler** [reply]
@primos63 great write up! Thanks, that helps with my understanding. Maybe colorScheme was the wrong prop to use as an example. I was just using colorScheme because it's in the example used in the [Chakra docs](https://chakra-ui.com/docs/theming/customize-theme#using-theme-extensions).

I can't get *any* of the extended theme options to work for Alerts. Here's an example for borderRadius, where the extended theme is not applied, but passing a prop does the trick:

```
const customTheme = extendTheme({
    components: {
        Alert: {
            baseStyle: {
                borderRadius: "8px"
            }
        }
    }
})

export const App = () => (
  <ChakraProvider theme={customTheme}>
      <Container>
          <VStack spacing={8} m={8}>
              
              <Alert>
                  <AlertIcon />
                  <AlertDescription>
                      I SHOULD have soft corners
                  </AlertDescription>
              </Alert>
              
              <Alert borderRadius="8px">
                  <AlertIcon />
                  <AlertDescription>
                      I DO have soft corners
                  </AlertDescription>
              </Alert>

          </VStack>
      </Container>
  </ChakraProvider>
)
```


![Screen Shot 2021-09-21 at 12 46 34 PM](https://user-images.githubusercontent.com/15269424/134237561-a3098456-3aa0-4c4e-8f91-860419ef6deb.png)

**Reply by primos63** [reply]
Alerts are multipart components. You have to refer to the theme source which is available from the docs for the component.
The parts for Alert are `container`, `title`, `description` and `icon`
```javascript
const baseStyle: PartsStyleObject<typeof parts> = {
  container: {
    px: 4,
    py: 3,
  },
  title: {
    fontWeight: "bold",
    lineHeight: 6,
    marginEnd: 2,
  },
  description: {
    lineHeight: 6,
  },
  icon: {
    flexShrink: 0,
    marginEnd: 3,
    w: 5,
    h: 6,
  },
}
```
So to affect the border you need to style the `container`
```jsx
components: {
    Alert: {
      baseStyle: {
        container: {
          borderRadius: '8px'
        }
      },
 ```

**Reply by georgeweiler** [reply]
amazing, that's exactly what I needed. Thank you!

**Reply by georgeweiler** [reply]
@primos63 What if you *did* want to adjust the colors of a specific status like the "info" status? I can see that chakra offers a "variants" key in theme config, but I can't find a way to adjust the styles of each status.

consider these design requirements, where the "info" status is white, not blue:
![Screen Shot 2021-09-22 at 2 53 01 PM](https://user-images.githubusercontent.com/15269424/134427132-27fa3fa9-531a-4129-aafa-e75d900bdc81.png)

editing baseStyles would work for borderRadius, but I'm unsure how I would adjust the colors for each status individually? 
In a perfect world I'd do something like:
```
const customTheme = extendTheme({
    components: {
        Alert: {
            statuses: {
                info: {
                    // info css adjustments
                },
                error: {
                    // error css adjustments 
                },
            }
        }
    }
})
```

**Reply by primos63** [reply]
Well, you'd go to my [CodeSandbox app](https://codesandbox.io/s/chakra-ui-theme-extension-w5u2n?file=/src/theme/Alert/index.js).

**Reply by georgeweiler** [reply]
that's a gold mine, thanks for sharing.
